src/glmtools.jl

```diff
@@ -98,3 +98,22 @@ loglik_obs(::Binomial, y, μ, wt, ϕ) = logpdf(Binomial(Int(wt), μ), Int(y*wt))
 loglik_obs(::Gamma, y, μ, wt, ϕ) = wt*logpdf(Gamma(1/ϕ, μ*ϕ), y)
 loglik_obs(::Normal, y, μ, wt, ϕ) = wt*logpdf(Normal(μ, sqrt(ϕ)), y)
 loglik_obs(::Poisson, y, μ, wt, ϕ) = wt*logpdf(Poisson(μ), y)
+
+# TODO: combine fweights, pweights, aweights
+function normalizeWeights(fweights::FrequencyWeights, pweights::ProbabilityWeights,
+                          aweights::AnalyticWeights)
+    nf = length(fweights)
+    np = length(pweights)
+    na = length(aweights)
+    if (nf != 0 && np == 0 && na == 0)
+        return fweights |> collect
+    elseif (nf == 0 && np != 0 && na == 0)
+        return pweights |> collect
+    elseif (nf == 0 && np == 0 && na != 0)
+        return aweights |> collect
+    elseif (nf != 0 && np != 0)
+        return (fweights .* pweights) |> collect
+    else
+        return zeros(Float64, 0)
```

Comment:
Actually the empty vector gets overridden at some point by `ones(n)`.  I haven't checked exactly where or why it was done.  I suspect it was so that a quantity could be normalized by `sum(weights)`.